src/SignalRServiceExtension/TriggerBindings/SignalRTriggerBinding.cs

```diff
@@ -0,0 +1,114 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Host.Bindings;
+using Microsoft.Azure.WebJobs.Host.Listeners;
+using Microsoft.Azure.WebJobs.Host.Protocols;
+using Microsoft.Azure.WebJobs.Host.Triggers;
+
+namespace Microsoft.Azure.WebJobs.Extensions.SignalRService
+{
+    internal class SignalRTriggerBinding : ITriggerBinding
+    {
+        private const string HubNameKey = "hubName";
+
+        private readonly ParameterInfo _parameterInfo;
+        private readonly SignalRTriggerAttribute _attribute;
+        private readonly SignalRTriggerRouter _router;
+
+        public SignalRTriggerBinding(ParameterInfo parameterInfo, SignalRTriggerAttribute attribute, SignalRTriggerRouter router)
+        {
+            _parameterInfo = parameterInfo ?? throw new ArgumentNullException(nameof(parameterInfo));
+            _attribute = attribute ?? throw new ArgumentNullException(nameof(attribute));
+            _router = router ?? throw new ArgumentNullException(nameof(router));
+        }
+
+        public Task<ITriggerData> BindAsync(object value, ValueBindingContext context)
+        {
+            var bindingData = new Dictionary<string, object>(StringComparer.OrdinalIgnoreCase);
+
+            if (value is SignalRTriggerEvent triggerEvent)
+            {
+                var bindingContext = triggerEvent.Context;
+                bindingData.Add(HubNameKey, bindingContext.HubName);
+                
+                return Task.FromResult<ITriggerData>(new TriggerData(new SignalRTriggerValueProvider(bindingContext), bindingData));
+            }
+
+            return Task.FromResult<ITriggerData>(null);
+        }
+
+        public Task<IListener> CreateListenerAsync(ListenerFactoryContext context)
+        {
+            if (context == null)
+            {
+                throw new ArgumentNullException(nameof(context));
+            }
+            
+            var functionNameAttribute = _parameterInfo.Member.GetCustomAttribute<FunctionNameAttribute>(false);
+            var methodName = functionNameAttribute.Name;
+
+            return Task.FromResult<IListener>(new SignalRListener(context.Executor, _router, _attribute.HubName, methodName));

```

Comment:
>SignalRListener [](start = 50, length = 15)

Looks like the key point is to add Excutor for a unique pair (hubname, methodname) into router, how about simple add Excutor to router and use an empty listner instead.